lib/Sema/CSApply.cpp

```diff
@@ -3414,6 +3414,14 @@ namespace {
                                              nameLoc.getBaseNameLoc(), toType));
       }
 
+      case OverloadChoiceKind::MaterializePack: {
+        auto packType = simplifyType(cs.getType(expr));
```

Comment:
Nit: there is `solution.getResolvedType(expr);` which was added relatively recently.